File: CS/UseFunctionCriteriaOperators.Module/DatabaseUpdate/Updater.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
//using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Security;
using UseFunctionCriteriaOperators.Module.BusinessObjects;

namespace UseFunctionCriteriaOperators.Module.DatabaseUpdate {
   public class Updater : ModuleUpdater {
      public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) {
      }
      public override void UpdateDatabaseAfterUpdateSchema() {
         base.UpdateDatabaseAfterUpdateSchema();

         DateTime todayDate = DateTime.Today;
         for (int i = -14; i < 5; i++ ) {
            Task task = ObjectSpace.FindObject<Task>(CriteriaOperator.Parse("DueDate = ADDDAYS(LocalDateTimeToday(), "+i+")"));
            if (task == null) {
               task = ObjectSpace.CreateObject<Task>();
               task.DueDate = todayDate.AddDays(i);
               task.Save();
            }
         }
      }
   }
}

File: VB/UseFunctionCriteriaOperators.Module/DatabaseUpdate/Updater.vb
Imports System

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
'using DevExpress.Persistent.BaseImpl;
Imports DevExpress.ExpressApp.Security
Imports UseFunctionCriteriaOperators.Module.BusinessObjects

Namespace UseFunctionCriteriaOperators.Module.DatabaseUpdate
   Public Class Updater
       Inherits ModuleUpdater

      Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
          MyBase.New(objectSpace, currentDBVersion)
      End Sub
      Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
         MyBase.UpdateDatabaseAfterUpdateSchema()

         Dim todayDate As Date = Date.Today
         For i As Integer = -14 To 4
            Dim task As Task = ObjectSpace.FindObject(Of Task)(CriteriaOperator.Parse("DueDate = ADDDAYS(LocalDateTimeToday(), " & i & ")"))
            If task Is Nothing Then
               task = ObjectSpace.CreateObject(Of Task)()
               task.DueDate = todayDate.AddDays(i)
               task.Save()
            End If
         Next i
      End Sub
   End Class
End Namespace
